Content.Shared/MobState/Components/MobStateComponent.cs

```diff
@@ -312,6 +313,18 @@ private void SetMobState(IMobState? old, (IMobState state, FixedPoint2 threshold
             state.EnterState(Owner);
             state.UpdateState(Owner, threshold);
 
+            if (old != null)
+            {
+               var stateChangeEvent = new MobStateChangedEvent(this, old, state);
+               Owner.EntityManager.EventBus.RaiseLocalEvent(OwnerUid, stateChangeEvent);
+            }
+            else
+            {
+                var stateChangeEvent = new MobStateChangedEvent(this, null, state);
+                Owner.EntityManager.EventBus.RaiseLocalEvent(OwnerUid, stateChangeEvent);
+            }
```

Comment:
Not using null state on this seems like a bad idea.

Reply:
What are you referring to?

Reply:
Just raise the event with old considering it's nullable anyway.